module-bsp/board/rt1051/bsp/eink/EinkDisplay.cpp

```diff
@@ -62,9 +130,62 @@ namespace service::eink
         delete[] currentWaveform.LUTDData;
     }
 
-    EinkStatus_e EinkDisplay::resetAndInit()
+    EinkStatus EinkDisplay::updateDisplay(std::uint8_t *frameBuffer, const EinkRefreshMode refreshMode)
     {
-        return EinkResetAndInitialize();
+        return update(frameBuffer);
+    }
+
+    EinkStatus EinkDisplay::refreshDisplay(const EinkRefreshMode refreshMode)
+    {
+        const auto isDeepRefresh = refreshMode == EinkRefreshMode::REFRESH_DEEP;
+        return refresh(isDeepRefresh ? EinkDisplayTimingsDeepCleanMode : EinkDisplayTimingsFastRefreshMode);
+    }
+
+    EinkStatus EinkDisplay::prepareDisplay(const EinkRefreshMode refreshMode, const WaveformTemperature behaviour)
+    {
+        powerOn();
+
+        const auto temperature =
+            behaviour == WaveformTemperature::KEEP_CURRENT ? getLastTemperature() : EinkGetTemperatureInternal();
+
+        if (refreshMode == EinkRefreshMode::REFRESH_DEEP) {
+            EinkStatus status;
+            status = setWaveform(EinkWaveforms_e::EinkWaveformGC16, temperature);
+            if (status == EinkStatus::EinkOK) {
+                dither();
+            }
+            return status;
+        }
```

Comment:
The `else` part is redundant.

Reply:
what do you mean?
behaviour in if base part and in else part is different, which part is redundant?

Reply:
I mean that lines 159 and 161 could be erased :)